dashboard/app/models/contact_rollups_processed.rb

```diff
@@ -92,11 +92,28 @@ def self.get_data_aggregation_query
     SQL
   end
 
-  # Parses a JSON string contains all data and metadata of a contact.
-  # It will throw exception if cannot parse the entire input string.
+  # Parses a JSON string containing contact data and metadata.
+  # Throw exception if cannot parse the entire input string.
   #
-  # @param [String] str represents a JSON array. Each array item is a hash {sources:String, data:Hash, data_updated_at:DateTime}.
-  # @return [Hash] a hash with string keys {table_name => {field_name => value}}
+  # @example
+  #   Input string:
+  #     '[{"sources":"table1", "data":{"opt_in":1}, "data_updated_at":"2020-06-16"}]'
```

Comment:
Is the input string actually:

`[{"s":"table1", "d":{"opt_in":1}, "u":"2020-06-16"}]`

Reply:
That's right. I'll explain that in the comment